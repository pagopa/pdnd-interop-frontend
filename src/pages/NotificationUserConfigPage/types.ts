import { type UserProductRole } from '@/types/party.types'

export type NotificationConfigType = 'email' | 'inApp'

export type NotificationSubSectionSchema = {
  name: string
  title: string
  components: {
    key: string
    title: string
    description: string
    visibility: UserProductRole[]
  }[]
}
export type NotificationSectionSchema = {
  title: string
  subsections: NotificationSubSectionSchema[]
}
export type NotificationConfigSchema = {
  [key: string]: NotificationSectionSchema
}
