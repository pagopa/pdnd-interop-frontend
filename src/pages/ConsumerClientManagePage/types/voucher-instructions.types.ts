import type { StepperStepComponentProps } from '@/types/common.types'
import type { PublicKeys } from '@/types/key.types'
import type { DecoratedPurpose } from '@/types/purpose.types'

export type VoucherInstructionsStepProps = StepperStepComponentProps & {
  clientId: string
  clientKeys: PublicKeys
  purpose?: DecoratedPurpose
  purposeId?: string
}
