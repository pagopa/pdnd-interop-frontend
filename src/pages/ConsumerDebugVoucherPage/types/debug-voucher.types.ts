import type { ClientKind } from '@/api/api.generatedTypes'

export type TokenGenerationValidationRequest = {
  client_id?: string
  client_assertion: string
  client_assertion_type: string
  grant_type: string
}

export type TokenGenerationValidationResult = {
  clientKind?: ClientKind
  steps: TokenGenerationValidationSteps
  eservice?: TokenGenerationValidationEService
}

export type TokenGenerationValidationSteps = {
  clientAssertionValidation: TokenGenerationValidationEntry
  publicKeyRetrieve: TokenGenerationValidationEntry
  clientAssertionSignatureVerification: TokenGenerationValidationEntry
  platformStatesVerification: TokenGenerationValidationEntry
}

export type TokenGenerationValidationEntry = {
  result: TokenGenerationValidationStepResult
  failures: Array<TokenGenerationValidationStepFailure>
}

export type TokenGenerationValidationStepResult = 'PASSED' | 'SKIPPED' | 'FAILED'

export type TokenGenerationValidationStepFailure = {
  code: string
  reason: string
}

export type TokenGenerationValidationEService = {
  id: string
  descriptorId: string
  version: string
  name: string
}
