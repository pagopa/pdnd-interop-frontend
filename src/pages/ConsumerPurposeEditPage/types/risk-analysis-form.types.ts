import type { FormConfigQuestion } from '@/api/api.generatedTypes'

export type AnswerValue = string | Array<string> | boolean
export type Questions = Record<string, FormConfigQuestion>
export type Answers = Record<string, AnswerValue>
