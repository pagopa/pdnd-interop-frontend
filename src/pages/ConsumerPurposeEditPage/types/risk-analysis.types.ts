import type { LangCode } from '@/types/common.types'
import type { TFunction } from 'i18next'
import type { UseFormReturn } from 'react-hook-form'

export type MultiLangEntry = {
  it: string
  en: string
}

export type Dependency = {
  id: string
  value: unknown
}

export type QuestionV1 = {
  id: string
  label: MultiLangEntry
  defaultValue: AnswerValue
  options?: Array<{
    label: MultiLangEntry
    value: string
  }>
  dependencies: Array<Dependency>
  type: 'text' | 'radio' | 'checkbox' | 'select-one'
  infoLabel?: MultiLangEntry
  required: boolean
}

export type QuestionV2 = {
  id: string
  /**
   * The HTML5 input type
   */
  type: 'text' | 'radio' | 'checkbox' | 'select-one' | 'switch'
  /**
   * Used for backend validation
   */
  dataType: 'single' | 'multi' | 'freeText'
  label: MultiLangEntry
  defaultValue: AnswerValue
  options?: Array<{
    label: MultiLangEntry
    value: string
  }>
  dependencies: Array<Dependency>
  /**
   * Declares the dependency of the state of one of its own option if the user
   * sets a specific value in another question.
   * EX:
   *
   * ```ts
   * {
   *   PREPARE_ICE_CREAM: [
   *     {
   *       id: "hasMilk",
   *       value: false,
   *     }
   *   ]
   * }
   * ```
   *
   * If the user sets the value false to the "hasMilk" question, the option "PREPARE_ICE_CREAN"
   * will be hidden
   *
   */
  hideOption?: Record<string, Array<Dependency>>
  infoLabel?: MultiLangEntry
  required: boolean
  validation?: {
    maxLength?: number
  }
}

export type Question = QuestionV1 | QuestionV2

export type RiskAnalysis = {
  version: string
  questions: Array<Question>
}

export type GetUpdatedQuestions = (
  values: Record<string, unknown>,
  riskAnalysis: RiskAnalysis
) => Questions

export type BuildForm = (
  questions: Questions,
  formMethods: UseFormReturn<Answers>,
  lang: LangCode,
  t: TFunction<'purpose'>
) => { formComponents: Array<React.ReactNode>; isSubmitBtnDisabled: boolean }

export type DynamicFormOperations = Record<
  string,
  {
    /**
     * Returns the updated question data.
     *
     * @param values - the actual form values
     * @param riskAnalysis - the risk analysis document
     *
     * @returns the updated questions data
     * */
    getUpdatedQuestions: GetUpdatedQuestions
    /**
     * Returns the updated form components.
     *
     * @param questions - the actual updated questions visible to the user
     * @param formik - formik object
     * @param lang - the actual active language
     * @param t - the TFunction of nexti18 internalization library
     *
     * @returns Array of components that should be rendered inside the form
     * */
    buildForm: BuildForm
  }
>

export type AnswerValue = string | Array<string> | boolean
export type Questions = Record<string, Question>
export type Answers = Record<string, AnswerValue>
