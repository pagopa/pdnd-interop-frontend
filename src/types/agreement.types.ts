import type {
  CertifiedTenantAttribute,
  DeclaredTenantAttribute,
  VerifiedTenantAttribute,
} from './attribute.types'
import type { DocumentRead } from './common.types'
import type { EServiceState } from './eservice.types'

export type AgreementState =
  | 'ACTIVE'
  | 'SUSPENDED'
  | 'PENDING'
  | 'ARCHIVED'
  | 'DRAFT'
  | 'REJECTED'
  | 'MISSING_CERTIFIED_ATTRIBUTES'

type AgreementProducer = {
  name: string
  id: string
}
type AgreementConsumer = {
  name: string
  selfcareId?: string
  externalId: { origin: string; value: string }
  createdAt: string
  updatedAt?: string
  id: string
  attributes: {
    certified: Array<CertifiedTenantAttribute>
    verified: Array<VerifiedTenantAttribute>
    declared: Array<DeclaredTenantAttribute>
  }
  contactMail: {
    address: string
    description?: string
  }
}

type AgreementEService = {
  id: string
  name: string
  version: string
  activeDescriptor?: {
    id: string
    state: EServiceState
    version: string
  }
}

export type AgreementAttribute = {
  id: string
  description: string
  name: string
  creationTime: string
}

export type AgreementListingItem = {
  id: string
  state: AgreementState
  consumer: {
    id: string
    name: string
  }
  eservice: {
    id: string
    name: string
    producer: {
      id: string
      name: string
    }
  }
  descriptor: {
    id: string
    state: EServiceState
    version: string
  }
  suspendedByConsumer?: boolean
  suspendedByProducer?: boolean
  suspendedByPlatform?: boolean
  canBeUpgraded: boolean
}

export type AgreementSummary = {
  id: string
  descriptorId: string
  consumer: AgreementConsumer
  producer: AgreementProducer
  state: AgreementState
  eservice: AgreementEService
  verifiedAttributes: Array<AgreementAttribute>
  certifiedAttributes: Array<AgreementAttribute>
  declaredAttributes: Array<AgreementAttribute>
  suspendedByProducer?: boolean
  suspendedByConsumer?: boolean
  suspendedByPlatform?: boolean
  consumerNotes?: string
  consumerDocuments: Array<DocumentRead & { createdAt: string }>
  createdAt: string
  updatedAt?: string
  rejectionReason?: string
  isContractPresent: boolean
}
