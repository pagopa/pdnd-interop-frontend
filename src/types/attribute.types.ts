import type { Attribute } from '@/api/api.generatedTypes'

export type AttributeKey = 'certified' | 'verified' | 'declared'

export type FrontendAttribute = {
  attributes: Array<Attribute>
  explicitAttributeVerification: boolean
}

export type FrontendAttributes = Record<AttributeKey, Array<FrontendAttribute>>

export type PartyAttributes = Record<AttributeKey, Array<PartyAttribute>>

export type AttributeState = 'ACTIVE' | 'REVOKED'

export type PartyAttribute = {
  id: string
  name: string
  state: AttributeState
}
