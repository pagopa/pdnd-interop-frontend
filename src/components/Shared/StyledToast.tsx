import React from 'react'
import { ToastProps } from '../../../types'
import { Alert, Snackbar } from '@mui/material'
import { useTranslation } from 'react-i18next'

export function StyledToast({ outcome, message, onClose, autoHideDuration }: ToastProps) {
  const { t } = useTranslation('shared-components', { keyPrefix: 'styledToast' })

  return (
    <Snackbar
      open={true}
      onClose={onClose}
      autoHideDuration={autoHideDuration}
      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
      sx={{ maxWidth: 480 }}
    >
      <Alert severity={outcome} onClose={onClose} variant="outlined">
        {message || t(`default.${outcome}.message`)}
      </Alert>
    </Snackbar>
  )
}
