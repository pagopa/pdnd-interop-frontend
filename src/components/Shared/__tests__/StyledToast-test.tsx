import { noop } from 'lodash'
import React from 'react'
import renderer from 'react-test-renderer'
import { ToastProps } from '../../../../types'
import { StyledToast } from '../StyledToast'

// For testing this component this way
// see this issue: https://github.com/mui-org/material-ui/issues/17119
describe('Snapshot', () => {
  it('matches success tooltip', () => {
    const props: ToastProps = { outcome: 'success', onClose: noop }
    let component
    renderer.act(() => {
      component = renderer.create(<StyledToast {...props} />, {
        createNodeMock: (node) => {
          return document.createElement(node.type as keyof HTMLElementTagNameMap)
        },
      })
    })
    const tree = (component as unknown as renderer.ReactTestRenderer).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches error tooltip', () => {
    const props: ToastProps = { outcome: 'error', onClose: noop }
    let component
    renderer.act(() => {
      component = renderer.create(<StyledToast {...props} />, {
        createNodeMock: (node) => {
          return document.createElement(node.type as keyof HTMLElementTagNameMap)
        },
      })
    })
    const tree = (component as unknown as renderer.ReactTestRenderer).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
