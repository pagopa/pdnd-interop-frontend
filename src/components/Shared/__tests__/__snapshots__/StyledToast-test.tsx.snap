// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches error tooltip 1`] = `
<div
  className="MuiSnackbar-root MuiSnackbar-anchorOriginBottomRight css-sfbrjq-MuiSnackbar-root"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-145dpyk-MuiPaper-root-MuiAlert-root"
    direction="up"
    role="alert"
    style={
      Object {
        "opacity": 1,
        "transform": "scale(1, 1)",
        "visibility": undefined,
      }
    }
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-acap47-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        Errore
      </div>
      Non è stato possibile completare l'operazione
    </div>
    <div
      className="MuiAlert-action css-ki1hdl-MuiAlert-action"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Close"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Snapshot matches success tooltip 1`] = `
<div
  className="MuiSnackbar-root MuiSnackbar-anchorOriginBottomRight css-sfbrjq-MuiSnackbar-root"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-13rvizp-MuiPaper-root-MuiAlert-root"
    direction="up"
    role="alert"
    style={
      Object {
        "opacity": 1,
        "transform": "scale(1, 1)",
        "visibility": undefined,
      }
    }
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="SuccessOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-acap47-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        Successo
      </div>
      L'operazione è andata a buon fine
    </div>
    <div
      className="MuiAlert-action css-ki1hdl-MuiAlert-action"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Close"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
